routers/user/profile.go

```diff
@@ -233,7 +233,8 @@ func Profile(ctx *context.Context) {
 		}
 	}
 
-	ctx.Data["ShowUserEmail"] = setting.UI.ShowUserEmail
+	ctx.Data["ShowUserEmail"] = (len(ctxUser.Email) > 0 && ctx.IsSigned && !ctxUser.KeepEmailPrivate) ||
```

Comment:
Condition should probably be:
```
len(ctxUser.Email) > 0 && ctx.IsSigned 
   && (!ctxUser.KeepEmailPrivate || ctxUser.ID == ctx.User.ID)
```

Reply:
I hope to get some free time tomorrow and test locally ( fix ) too

Reply:
My provided condition should work also for unauthorized users